mne/annotations.py

```diff
@@ -880,6 +910,29 @@ def _check_event_id(event_id, raw):
         raise ValueError('Invalid input event_id')
 
 
+def _check_event_description(event_desc, events):
+    """Check event_id and convert to default format."""
+    # check out event_id dict
+    if event_desc is None:  # convert to int to make typing-checks happy
+        event_desc = list(np.unique(events[:, 2]))
+
+    if isinstance(event_desc, np.ndarray):
+        if event_desc.ndim == 1:
+            event_desc = list(event_desc)
```

Comment:
curent code will fail if you pass an array of ndim > 1.

what I would do is cast to array anything that is not a dict and callable with np.assarray and then assert that ndim == 1

Reply:
Fine, but if I do an assertion here, a 2d array won't hit the nice `ValueError` message you suggest below :)